lib/full-site-editing/template-parts.php

```diff
@@ -118,3 +147,38 @@ function set_unique_slug_on_create_template_part( $post_id ) {
 	}
 }
 add_action( 'save_post_wp_template_part', 'set_unique_slug_on_create_template_part' );
+
+/**
+ * Returns a filtered list of allowed section types for template parts.
+ *
+ * @return array The supported template part section types.
+ */
+function gutenberg_get_allowed_template_part_section_types() {
+	$default_section_types = array(
+		'header',
+		'footer',
+		'other',
+	);
+
+	/**
+	 * Filters the list of allowed template part section types.
+	 *
+	 * @param array $default_section_types An array of supported section types.
+	 */
+	return apply_filters( 'default_template_part_section_types', $default_section_types );
+}
+
+/**
+ * Checks whether the input 'type' is a supported section type.
+ * Returns the input if supported, otherwise returns the 'other' type.
+ *
+ * @param string $type Template part section name.
+ *
+ * @return string Input if supported, else 'other'.
+ */
+function gutenberg_filter_template_part_section_type( $type ) {
+	if ( in_array( $type, gutenberg_get_allowed_template_part_section_types(), true ) ) {
+		return $type;
+	}
+	return 'other';
```

Comment:
What are the benefits of defaulting un-allowed types to `other` instead of leaving them undefined and maybe showing an error message or a warning?

Reply:
Maybe its not necessary, but it allows a distinction between those that are uncategorized vs. those that are intended for a specific purpose but categorized by an unsupported term -  template parts who had no section defined at all `null`/`undefined` vs. template parts who are intended for a particular section that happens to be unsupported `'other'`.

I suppose we should probably trigger a warning here as well?

Reply:
Set up a warning for this - 8c5f17b